lib/web/service-worker/service-worker-global-scope.js

```diff
@@ -0,0 +1,84 @@
+import { parentPort } from 'node:worker_threads'
+import { Clients, kAddClient } from './clients.js'
+import { ServiceWorker } from './service-worker.js'
+import { Client } from './client.js'
+import { CacheStorage } from '../cache/cachestorage.js'
+
+/**
+ * Custom implementation of the `ServiceWorkerGlobalScope` object.
+ * This acts as `self` in the global scope of the Service Worjer.
+ */
+export class ServiceWorkerGlobalScope extends EventTarget {
+  /** @type {import('./service-worker-container.js').WorkerData} */
+  #parentData
+
+  /**
+   * @param {import('./service-worker-container.js').WorkerData} parentData
+   */
+  constructor (parentData) {
```

Comment:
## Todo

Implement the `registration` representation on the global scope. Something similar to how the `self.serviceWorker` is implemented. 

Those will be different objects in the main thread and in the worker thread but connected via messaging.